### PR TITLE
feat: added buildAuthorizeUrl and buildLogoutUrl from auth0-spa-js

### PR DESCRIPTION
**Pull request by THISS**
By submitting a PR to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/auth0/.github/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

Added in the buildAuthorizeUrl and buildLogoutUrl which were both in the https://github.com/auth0/auth0-spa-js repo. This helps with projects that need more management over the browser client eg. Ionic using capacitor requires a native "Browser" plugin to do redirects in app.

### References

Related PRs from the spa-js repo: https://github.com/auth0/auth0-spa-js/pull/280, https://github.com/auth0/auth0-spa-js/pull/595

### Testing

Added to the `auth-provider.test.tsx` unit tests a couple of noop tests for coverage that just make sure the call is passed through to the Auth0Client similar to a couple of other tests. I looked at cypress but it seems there is just a bare bones smoke test so didn't feel this needed the full treatment of getting the strings. Let me know if you would like them added

- [x] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `master`

### CODE REVIEW

File: src/auth0-provider.tsx [hunk 1]
```diff
@@ -228,6 +229,17 @@ const Auth0Provider = (opts: Auth0ProviderOptions): JSX.Element => {
     })();
   }, [client, onRedirectCallback, skipRedirectCallback]);
 
+  const buildAuthorizeUrl = useCallback(
+    (opts?: Auth0RedirectLoginOptions): Promise<string> =>
```
Comment (adamjmcgrath) [inline src/auth0-provider.tsx:233]:
```js
const buildAuthorizeUrl = useCallback(
  (opts?: RedirectLoginOptions): Promise<string> =>
    client.buildAuthorizeUrl(toAuth0LoginRedirectOptions(opts)),
  [client]
);
```

(this SDK uses `redirectUri` and spa-js uses `redirect_uri`)

File: src/auth0-provider.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import {
   CacheLocation,
   IdToken,
   LogoutOptions,
+  LogoutUrlOptions,
```
Comment (adamjmcgrath) [inline src/auth0-provider.tsx:8]: Can you export this from https://github.com/auth0/auth0-react/blob/master/src/index.tsx#L17-L26

**Review by adamjmcgrath — CHANGES_REQUESTED** [review]

File: src/auth0-provider.tsx (hunk 1 above)
Comment (THISS) [reply src/auth0-provider.tsx:233]: ah, good catch 👍 thanks for that man

File: src/auth0-provider.tsx (hunk 2 above)
Comment (THISS) [reply src/auth0-provider.tsx:8]: apologies on the delay between me seeing this and your review.

File: src/auth0-provider.tsx [hunk 3]
```diff
@@ -228,6 +229,17 @@ const Auth0Provider = (opts: Auth0ProviderOptions): JSX.Element => {
     })();
   }, [client, onRedirectCallback, skipRedirectCallback]);
 
+  const buildAuthorizeUrl = useCallback(
+    (opts?: RedirectLoginOptions): Promise<string> =>
+      client.buildAuthorizeUrl(opts),
```
Comment (adamjmcgrath) [inline src/auth0-provider.tsx:234]: Annoyingly you still need to do `client.buildAuthorizeUrl(toAuth0LoginRedirectOptions(opts))`

Somewhere in between making spa js and auth0-react we changed the options from snake case to camel case

Comment (THISS) [reply src/auth0-provider.tsx]: sorry about that, I had checked how it is converted but forgot to add in the method that converts it...

**Review by adamjmcgrath — APPROVED** [review]

**Comment by adamjmcgrath** [comment]
Thanks @THISS!

**Comment by aaronadamsCA** [comment]
I was literally just opening a feature request to expose the logout URL, and I was growing increasingly confused why I could see this capability on GitHub, but not in my IDE. Imagine my surprise that it was added within the last 24 hours.

Thanks for this! Putting it to use already in a Shopify app, which runs inside an iframe and therefore can't just call `logout()`. 🙌

**Comment by adamjmcgrath** [comment]
Release in https://github.com/auth0/auth0-react/releases/tag/v1.3.0

**Pull request merged**